src/main/java/wooteco/subway/service/LineService.java

```diff
@@ -3,27 +3,46 @@
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 import wooteco.subway.dao.LineDao;
+import wooteco.subway.dao.SectionDao;
+import wooteco.subway.dao.StationDao;
 import wooteco.subway.domain.Line;
+import wooteco.subway.domain.Section;
+import wooteco.subway.domain.Station;
 import wooteco.subway.dto.LineRequest;
 import wooteco.subway.dto.LineResponse;
+import wooteco.subway.dto.SectionRequest;
+import wooteco.subway.dto.StationResponse;
 
-import java.util.List;
+import java.util.*;
+import java.util.function.Consumer;
 import java.util.stream.Collectors;
 
 @Transactional
 @Service
 public class LineService {
     private final LineDao lineDao;
+    private final StationDao stationDao;
+    private final SectionDao sectionDao;
 
-    public LineService(LineDao lineDao) {
+    public LineService(LineDao lineDao, StationDao stationDao, SectionDao sectionDao) {
         this.lineDao = lineDao;
+        this.stationDao = stationDao;
+        this.sectionDao = sectionDao;
     }
 
     public LineResponse saveLine(LineRequest lineRequest) {
         checkExistLineByName(lineRequest);
-        final Line line = new Line(lineRequest.getName(), lineRequest.getColor());
-        final Long newLineId = lineDao.save(line);
-        return new LineResponse(newLineId, line.getName(), line.getColor());
+        final Long id = lineDao.save(lineRequest);
+        final Section section = saveInitialSection(id, lineRequest.getUpStationId(), lineRequest.getDownStationId(),
+                lineRequest.getDistance());
+        final Line line = new Line(id, lineRequest.getName(), lineRequest.getColor(), section);
+
+        final Station upStation = findByStationId(section.getUpStationId());
+        final Station downStation = findByStationId(section.getDownStationId());
+        final StationResponse upStationResponse = new StationResponse(upStation.getId(), upStation.getName());
+        final StationResponse downStationResponse = new StationResponse(downStation.getId(), downStation.getName());
+
+        return new LineResponse(id, line.getName(), line.getColor(), List.of(upStationResponse, downStationResponse));
```

Comment:
복잡도를 줄여보면 어떨까요?
```
validate(request);
Line line = createLine(request);
Section section = createSection(line);
return new LineResponse(line, section);
```
위와 같이 그룹핑을 할 수 있을거같은데 한번 고민해보셔요!

Reply:
현재 `Section` 도메인엔 역이 아닌 `역의 id`를 가지고 있는 상태라서 제시해주신 것처럼 바로 그룹핑을 하긴 어려웠습니다 😢 
대신! `createLine` 메서드로 그룹핑해주었고, `Station` 도메인을 `StationResponse` 로 변환해주는 `makeStationResponseByStation` 메서드가 원래 존재했었기 때문에 이를 재활용하여 아래 코드로 변경했습니다!
```suggestion
        checkExistLineByName(lineRequest);
        final Line line = createLine(lineRequest);

        final Station upStation = findByStationId(lineRequest.getUpStationId());
        final Station downStation = findByStationId(lineRequest.getDownStationId());
        final List<StationResponse> stationResponses = makeStationResponseByStation(List.of(upStation, downStation));

        return new LineResponse(line.getId(), line.getName(), line.getColor(), stationResponses);
```